src/Components/Shared/src/WebEventData.cs

```diff
@@ -245,4 +258,22 @@ private static ChangeEventArgs DeserializeChangeEventArgs(string eventArgsJson)
             return changeArgs;
         }
     }
+
+    [JsonSerializable(typeof(WebEventDescriptor), GenerationMode = JsonSourceGenerationMode.Serialization)]
```

Comment:
@layomia - using `JsonSourceGenerationMode.Serialization` means we only generate the "fast path" and not the "metadata".

What does this mean when we only support serialization right now, and not deserialization? Does it mean for deserialization it falls back to Reflection.Emit based deserialization?

Reply:
To use generated code for deserialization and avoiding Ref.Emit today, the setting would need to be `JsonSourceGenerationMode.MetadataAndSerialization`. The serializer would fallback to generated code that references setters and getters directly.

To make configuration here cleaner, we could change the enum to a flags enum:

```cs
[Flags]
public enum JsonSourceGenerationMode
{
  Unspecified = 0,
  Metadata = 1,
  Serialization = 2,
  Deserialization = 4, // Future
}
```

When we have fast-path logic for both read and write, the preferred setting for simple scenarios would be `Serialization | Deserialization`.